src/main/scala/is/hail/variant/Genotype.scala

```diff
@@ -176,6 +176,32 @@ class RowGenotype(r: Row) extends Genotype {
 }
 
 object Genotype {
+  def buildGenotypeExtractor(t: Type): (Any) => Genotype = {
+    if (t == TGenotype)
+      (a: Any) => a.asInstanceOf[Genotype]
+    else {
+      assert(t.isInstanceOf[TStruct])
+
+      val (gtt, gtq0) = t.queryTyped("GT")
+      val gtq = if (gtt == TCall) gtq0 else (a: Any) => null
+
+      val (adt, adq0) = t.queryTyped("AD")
+      val adq = if (adt == TArray(TInt)) adq0 else (a: Any) => null
+
+      val (dpt, dpq0) = t.queryTyped("DP")
+      val dpq = if (dpt == TArray(TInt)) dpq0 else (a: Any) => null
+
+      val (gqt, gqq0) = t.queryTyped("GQ")
+      val gqq = if (gqt == TInt) gqq0 else (a: Any) => null
+
+      val (plt, plq0) = t.queryTyped("PL")
+      val plq = if (plt == TInt) plq0 else (a: Any) => null
+
+      (a: Any) =>
+        new RowGenotype(Row(gtq(a), adq(a), dpq(a), gqq(0), plq(0)))
```

Comment:
RowGenotype will fail if someone calls `_fakeRef` or `_isLinearScale`. Need to define 6th and 7th element in row.